### PR TITLE
[enhance] demo for expense claim #5731

### PR DESCRIPTION
**Pull request by saurabh6790**
- make expense claim
- mark some as approved
- make payment against some of approved
- mark some as rejected
- keep some in draft state

### CODE REVIEW

File: erpnext/hr/doctype/expense_claim_type/expense_claim_type.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+frappe.ui.form.on("Expense Claim Type", "onload", function(frm, dt, dn){
+	frm.fields_dict["default_account"].get_query = function(doc) {
+	return {
+		filters:{
+			"is_group": 0
```
Comment (nabinhait) [inline erpnext/hr/doctype/expense_claim_type/expense_claim_type.js]: "root_type": "Expense",
"company": frm.doc.company

**Comment by saurabh6790** [comment]
@nabin Fixed.

File: erpnext/demo/user/hr.py [hunk 2]
```diff
@@ -30,6 +31,61 @@ def work():
...
+		expense_claim.extend('expenses', get_expenses())
+		expense_claim.employee = get_random("Employee")
+		expense_claim.company = frappe.flags.company
+		expense_claim.posting_date = frappe.flags.current_date
+		expense_claim.exp_approver = filter((lambda x: x[0] != 'Administrator'), get_expense_approver(None, '', None, 0, 20, None))[0][0]
+		expense_claim.insert()
+
+		rand = random.random()
+
+		if rand >= 0.3 and rand <= 0.6 :
```
Comment (rmehta) [inline erpnext/demo/user/hr.py]: can be just `< 0.3`

File: erpnext/hr/doctype/expense_claim/expense_claim.py [hunk 3]
```diff
@@ -68,3 +73,48 @@ def get_expense_approver(doctype, txt, searchfield, start, page_len, filters):
...
 	""", ("%" + txt + "%"))
+
+@frappe.whitelist()
+def make_bank_entry(docname):
+	from erpnext.accounts.doctype.journal_entry.journal_entry import get_default_bank_cash_account
+
+	expense_claim = frappe.get_doc("Expense Claim", docname)
+	default_bank_cash_account = get_default_bank_cash_account(expense_claim.company, "Bank")
+
+	je = frappe.new_doc("Journal Entry")
```
Comment (rmehta) [inline erpnext/hr/doctype/expense_claim/expense_claim.py]: Make Payment Entry!

File: erpnext/hr/doctype/expense_claim_type/expense_claim_type.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
+// License: GNU General Public License v3. See license.txt
+
+cur_frm.set_query("default_account", "accounts", function(doc, cdt, cdn) {
```
Comment (rmehta) [inline erpnext/hr/doctype/expense_claim_type/expense_claim_type.js]: cur_frm 😢

File: erpnext/hr/doctype/expense_claim/expense_claim.py (hunk 3 above)
Comment (nabinhait) [inline erpnext/hr/doctype/expense_claim/expense_claim.py]: @rmehta As of now, Payment Entry cannot be made against Expense Claim, will implement later.

Comment (rmehta) [inline erpnext/hr/doctype/expense_claim/expense_claim.py]: @nabinhait how difficult?

Comment (nabinhait) [inline erpnext/hr/doctype/expense_claim/expense_claim.py]: @rmehta Did not give enough thought. But everything is hard-coded based on party type, and for this we need to introduce a new party type Employee.

**Comment by saurabh6790** [comment]
@rmehta Fixed!

**Pull request merged**